src/assets/sql/stored-procedure/countAppointmentWithUserIdAndServiceIdAndPatientId.sql

```diff
@@ -0,0 +1,69 @@
+-------------------------------------------------------------------------- STORED PROCEDURE SCRIPT----------------------------------------------------------------
+DELIMITER $$
+DROP PROCEDURE IF EXISTS `countAppointmentsByCondition` $$
+CREATE DEFINER=`root`@`localhost` PROCEDURE `countAppointmentsByCondition`(IN `idUserIntegrated` VARCHAR(200), 
+                                                                           IN `idServiceIntegrated` VARCHAR(200), 
+                                                                           IN `idPatientIntegrated` VARCHAR(200), 
+                                                                           IN `startDate` VARCHAR(200), 
+                                                                           IN `endDate` VARCHAR(200))
+BEGIN
+	Declare startOffset Integer;
+    #-- Prepare pre-condition before running main query
+    IF idServiceIntegrated <> '' THEN
+        SET idServiceIntegrated = (SELECT ea_services.id from ea_services WHERE ea_services.id_integrated = idServiceIntegrated);
+    END IF;
+
+    IF idUserIntegrated <> '' THEN
+        SET idUserIntegrated = (SELECT ea_users.id from ea_users WHERE ea_users.id_integrated = idUserIntegrated);
+    END IF;
+
+    IF idPatientIntegrated <> '' THEN
+        SET idPatientIntegrated = (SELECT ea_users.id from ea_users WHERE ea_users.id_integrated = idPatientIntegrated);
+    END IF;
+
+
+    # -- Start build final Query with parameter
+    SET @finalQuery = 'SELECT COUNT(ea_appointments.id) as "total" FROM ea_appointments WHERE 1 = 1';
```

Comment:
remove "WHERE 1 = 1"

Reply:
could not remove WHERE 1 =1 , this 1 =1 is a trick to concentrate multiple condition base on parameter. In case you don't all null parameter, you can be able to get all records with pagination.

Reply:
Yes, understood. I am not in favour of applying tricky code if there is no specific reason :)
You could drive it by building a separate where statement, at the end if the where statement ís not empty, lets concatenate the query and the where statement.
in case no parameters provided, the query to be executed without where statement

Reply:
@chicanh: I tried my best could not find out solution to concentrate where condition without `1=1`, please help me write the query which satisfied all following condition :(.

We have totally 5 params for WHERE condition: 
- idUserIntegrated, IdServiceIntegrated, IdPatientIntegrated, startDate, endDate

**Case 1:** 
- idUserIntegrated = null 
- IdServiceIntegrated = valid value
- IdPatientIntegrated = valid value
- startDate = valid value
- endDate = valid value

**Case 2:** 
- idUserIntegrated = valid value
- IdServiceIntegrated= null
- IdPatientIntegrated = valid value
- startDate = valid value
- endDate = valid value


**Case 3:** 
- idUserIntegrated = valid value
- IdServiceIntegrated= valid value
- IdPatientIntegrated = null
- startDate = valid value
- endDate = valid value

**Case 4:** 
- idUserIntegrated = valid value
- IdServiceIntegrated= valid value
- IdPatientIntegrated = valid value
- startDate = null
- endDate = valid value


**Case 5:** 
- idUserIntegrated = valid value
- IdServiceIntegrated= valid value
- IdPatientIntegrated = valid value
- startDate = valid value
- endDate = null

**Case 6:** 
- idUserIntegrated = null
- IdServiceIntegrated= null
- IdPatientIntegrated = null
- startDate = null
- endDate = null

Reply:
OK see, better to keep it there thanks